R/integrate_sces.R

```diff
@@ -172,17 +187,28 @@ integrate_harmony <- function(merged_sce,
     stop("The provided covariate columns are not all present in the `merged_sce` colData.")
   }
 
+  # check that the lambda values match up
+  if(length(covariate_cols) != length(covariate_lambda)){
+    stop("The number of covariate columns must be equal to the number of covariate lambda values.")
+  }
+
   # Setup harmony metadata
   covariate_cols <- c(batch_column, covariate_cols)
-  harmony_metadata <- tibble::as_tibble(colData(merged_sce)) %>%
+
+  harmony_metadata <- tibble::as_tibble(colData(merged_sce)) |>
     dplyr::select(
       dplyr::all_of(covariate_cols)
     )
 
+  # get a vector of all lambdas
+  lambda_vec <- c(batch_lambda, covariate_lambda)
+
   # Perform integration
-  harmony_results <- harmony::HarmonyMatrix(reducedDim(merged_sce, "PCA"),
+  harmony_results <- harmony::RunHarmony(
+    data_mat = reducedDim(merged_sce, "PCA"),
     meta_data = harmony_metadata,
```

Comment:
Is there any reason not to directly use `merged_sce` here?

If we do that we can remove the `harmony_metadata` definition as well, as it will get those automatically.
```suggestion
    merged_sce,
```
```